### ISSUE TITLE
`bundle install` does not respect lock file

### ISSUE THREAD

**Issue by tarbayev** [rubygems/rubygems#4122]
### Describe the problem as clearly as you can

After installing bundler 2.2.0 it stopped respecting `Gemfile.lock` because of platform mismatching platforms.
This leads to unexpected failure in our CI.

### Post steps to reproduce the problem

Run `gem install bundler`
Ensure that `bundle --version` prints 2.2.0

### Which command did you run?

`bundle install` contains the following message:
`Found changes from the lockfile, re-resolving dependencies because you added a new platform to your gemfile`
and ignores `Gemfile.lock`, by running bundler 2.2.0 instead of 2.1.4

`bundle platform` prints:
```
Your platform is: x86_64-darwin18

Your app has gems that work on these platforms:
* ruby
* x86_64-darwin-18

Your Gemfile does not specify a Ruby version requirement.
```

With bundler 2.1.4 `bundle platform` prints only `ruby` platform:
```
Your platform is: x86_64-darwin19

Your app has gems that work on these platforms:
* ruby

Your Gemfile does not specify a Ruby version requirement.
```

### What were you expecting to happen?

No changes to the `Gamefile.lock`

### What actually happened?

`Gemfile.lock` is updated with extra platform in addition to `ruby`:
```
 PLATFORMS
   ruby
+  x86_64-darwin-18
```

## Environment

```
Bundler       2.2.0
  Platforms   ruby, x86_64-darwin-18
Ruby          2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin18]
  Full Path   /Users/jenkins/.rbenv/versions/2.5.1/bin/ruby
  Config Dir  /Users/jenkins/.rbenv/versions/2.5.1/etc
RubyGems      3.0.6
  Gem Home    /Users/jenkins/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0
  Gem Path    /Users/jenkins/.gem/ruby/2.5.0:/Users/jenkins/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0
  User Home   /Users/jenkins
  User Path   /Users/jenkins/.gem/ruby/2.5.0
  Bin Dir     /Users/jenkins/.rbenv/versions/2.5.1/bin
Tools
  Git         2.25.1
  RVM         not installed
  rbenv       rbenv 1.1.2
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2020-12-10
Git SHA           d3628e5019
Released Version  true
```

### Gemfile

```ruby
source "https://rubygems.org"

gem 'cocoapods', '~> 1.8'
gem 'fastlane', '~> 2.0'
gem 'danger-gitlab', '~> 6.0.0'
gem 'danger-swiftlint', '~> 0.24.0'
gem 'danger-swiftformat', '~> 0.7.0'
gem 'danger-commit_lint'
gem 'rubocop', '~> 0.85.0'

gem 'xcode-install', '~> 2.6.0'
```

### fastlane/Pluginfile

```ruby
# Autogenerated by fastlane
#
# Ensure this file is checked in to source control!

gem 'fastlane-plugin-simctl'
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (3.0.2)
    activesupport (4.2.11.1)
      i18n (~> 0.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    addressable (2.7.0)
      public_suffix (>= 2.0.2, < 5.0)
    algoliasearch (1.27.1)
      httpclient (~> 2.8, >= 2.8.3)
      json (>= 1.5.1)
    ast (2.4.1)
    atomos (0.1.3)
    aws-eventstream (1.1.0)
    aws-partitions (1.371.0)
    aws-sdk-core (3.107.0)
      aws-eventstream (~> 1, >= 1.0.2)
      aws-partitions (~> 1, >= 1.239.0)
      aws-sigv4 (~> 1.1)
      jmespath (~> 1.0)
    aws-sdk-kms (1.38.0)
      aws-sdk-core (~> 3, >= 3.99.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-s3 (1.81.0)
      aws-sdk-core (~> 3, >= 3.104.3)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.1)
    aws-sigv4 (1.2.2)
      aws-eventstream (~> 1, >= 1.0.2)
    babosa (1.0.3)
    claide (1.0.3)
    claide-plugins (0.9.2)
      cork
      nap
      open4 (~> 1.3)
    cocoapods (1.8.4)
      activesupport (>= 4.0.2, < 5)
      claide (>= 1.0.2, < 2.0)
      cocoapods-core (= 1.8.4)
      cocoapods-deintegrate (>= 1.0.3, < 2.0)
      cocoapods-downloader (>= 1.2.2, < 2.0)
      cocoapods-plugins (>= 1.0.0, < 2.0)
      cocoapods-search (>= 1.0.0, < 2.0)
      cocoapods-stats (>= 1.0.0, < 2.0)
      cocoapods-trunk (>= 1.4.0, < 2.0)
      cocoapods-try (>= 1.1.0, < 2.0)
      colored2 (~> 3.1)
      escape (~> 0.0.4)
      fourflusher (>= 2.3.0, < 3.0)
      gh_inspector (~> 1.0)
      molinillo (~> 0.6.6)
      nap (~> 1.0)
      ruby-macho (~> 1.4)
      xcodeproj (>= 1.11.1, < 2.0)
    cocoapods-core (1.8.4)
      activesupport (>= 4.0.2, < 6)
      algoliasearch (~> 1.0)
      concurrent-ruby (~> 1.1)
      fuzzy_match (~> 2.0.4)
      nap (~> 1.0)
    cocoapods-deintegrate (1.0.4)
    cocoapods-downloader (1.2.2)
    cocoapods-plugins (1.0.0)
      nap
    cocoapods-search (1.0.0)
    cocoapods-stats (1.1.0)
    cocoapods-trunk (1.4.1)
      nap (>= 0.8, < 2.0)
      netrc (~> 0.11)
    cocoapods-try (1.1.0)
    colored (1.2)
    colored2 (3.1.2)
    commander-fastlane (4.4.6)
      highline (~> 1.7.2)
    concurrent-ruby (1.1.5)
    cork (0.3.0)
      colored2 (~> 3.1)
    danger (5.16.1)
      claide (~> 1.0)
      claide-plugins (>= 0.9.2)
      colored2 (~> 3.1)
      cork (~> 0.1)
      faraday (~> 0.9)
      faraday-http-cache (~> 1.0)
      git (~> 1.5)
      kramdown (~> 1.5)
      no_proxy_fix
      octokit (~> 4.7)
      terminal-table (~> 1)
    danger-commit_lint (0.0.7)
      danger-plugin-api (~> 1.0)
    danger-gitlab (6.0.0)
      danger (~> 5.0)
      gitlab (~> 4.2, >= 4.2.0)
    danger-plugin-api (1.0.0)
      danger (> 2.0)
    danger-swiftformat (0.7.0)
      danger-plugin-api (~> 1.0)
    danger-swiftlint (0.24.4)
      danger
      rake (> 10)
      thor (~> 0.19)
    declarative (0.0.20)
    declarative-option (0.1.0)
    digest-crc (0.6.1)
      rake (~> 13.0)
    domain_name (0.5.20190701)
      unf (>= 0.0.5, < 1.0.0)
    dotenv (2.7.6)
    emoji_regex (3.0.0)
    escape (0.0.4)
    excon (0.76.0)
    faraday (0.17.3)
      multipart-post (>= 1.2, < 3)
    faraday-cookie_jar (0.0.7)
      faraday (>= 0.8.0)
      http-cookie (~> 1.0.0)
    faraday-http-cache (1.3.1)
      faraday (~> 0.8)
    faraday_middleware (0.14.0)
      faraday (>= 0.7.4, < 1.0)
    fastimage (2.2.0)
    fastlane (2.152.0)
      CFPropertyList (>= 2.3, < 4.0.0)
      addressable (>= 2.3, < 3.0.0)
      aws-sdk-s3 (~> 1.0)
      babosa (>= 1.0.3, < 2.0.0)
      bundler (>= 1.12.0, < 3.0.0)
      colored
      commander-fastlane (>= 4.4.6, < 5.0.0)
      dotenv (>= 2.1.1, < 3.0.0)
      emoji_regex (>= 0.1, < 4.0)
      excon (>= 0.71.0, < 1.0.0)
      faraday (>= 0.17, < 1.0)
      faraday-cookie_jar (~> 0.0.6)
      faraday_middleware (>= 0.13.1, < 2.0)
      fastimage (>= 2.1.0, < 3.0.0)
      gh_inspector (>= 1.1.2, < 2.0.0)
      google-api-client (>= 0.37.0, < 0.39.0)
      google-cloud-storage (>= 1.15.0, < 2.0.0)
      highline (>= 1.7.2, < 2.0.0)
      json (< 3.0.0)
      jwt (>= 2.1.0, < 3)
      mini_magick (>= 4.9.4, < 5.0.0)
      multipart-post (~> 2.0.0)
      plist (>= 3.1.0, < 4.0.0)
      rubyzip (>= 2.0.0, < 3.0.0)
      security (= 0.1.3)
      simctl (~> 1.6.3)
      slack-notifier (>= 2.0.0, < 3.0.0)
      terminal-notifier (>= 2.0.0, < 3.0.0)
      terminal-table (>= 1.4.5, < 2.0.0)
      tty-screen (>= 0.6.3, < 1.0.0)
      tty-spinner (>= 0.8.0, < 1.0.0)
      word_wrap (~> 1.0.0)
      xcodeproj (>= 1.13.0, < 2.0.0)
      xcpretty (~> 0.3.0)
      xcpretty-travis-formatter (>= 0.0.3)
    fastlane-plugin-simctl (0.2.1)
      simctl (~> 1.6.5)
    fourflusher (2.3.1)
    fuzzy_match (2.0.4)
    gh_inspector (1.1.3)
    git (1.7.0)
      rchardet (~> 1.8)
    gitlab (4.8.0)
      httparty (>= 0.14.0)
      terminal-table (>= 1.5.1)
    google-api-client (0.38.0)
      addressable (~> 2.5, >= 2.5.1)
      googleauth (~> 0.9)
      httpclient (>= 2.8.1, < 3.0)
      mini_mime (~> 1.0)
      representable (~> 3.0)
      retriable (>= 2.0, < 4.0)
      signet (~> 0.12)
    google-cloud-core (1.5.0)
      google-cloud-env (~> 1.0)
      google-cloud-errors (~> 1.0)
    google-cloud-env (1.3.3)
      faraday (>= 0.17.3, < 2.0)
    google-cloud-errors (1.0.1)
    google-cloud-storage (1.28.0)
      addressable (~> 2.5)
      digest-crc (~> 0.4)
      google-api-client (~> 0.33)
      google-cloud-core (~> 1.2)
      googleauth (~> 0.9)
      mini_mime (~> 1.0)
    googleauth (0.13.1)
      faraday (>= 0.17.3, < 2.0)
      jwt (>= 1.4, < 3.0)
      memoist (~> 0.16)
      multi_json (~> 1.11)
      os (>= 0.9, < 2.0)
      signet (~> 0.14)
    highline (1.7.10)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    httparty (0.16.3)
      mime-types (~> 3.0)
      multi_xml (>= 0.5.2)
    httpclient (2.8.3)
    i18n (0.9.5)
      concurrent-ruby (~> 1.0)
    jmespath (1.4.0)
    json (2.3.1)
    jwt (2.2.2)
    kramdown (1.17.0)
    memoist (0.16.2)
    mime-types (3.3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2020.0512)
    mini_magick (4.10.1)
    mini_mime (1.0.2)
    minitest (5.13.0)
    molinillo (0.6.6)
    multi_json (1.15.0)
    multi_xml (0.6.0)
    multipart-post (2.0.0)
    nanaimo (0.3.0)
    nap (1.1.0)
    naturally (2.2.0)
    netrc (0.11.0)
    no_proxy_fix (0.1.2)
    octokit (4.18.0)
      faraday (>= 0.9)
      sawyer (~> 0.8.0, >= 0.5.3)
    open4 (1.3.4)
    os (1.1.1)
    parallel (1.19.1)
    parser (2.7.1.3)
      ast (~> 2.4.0)
    plist (3.5.0)
    public_suffix (4.0.6)
    rainbow (3.0.0)
    rake (13.0.1)
    rchardet (1.8.0)
    regexp_parser (1.7.1)
    representable (3.0.4)
      declarative (< 0.1.0)
      declarative-option (< 0.2.0)
      uber (< 0.2.0)
    retriable (3.1.2)
    rexml (3.2.4)
    rouge (2.0.7)
    rubocop (0.85.1)
      parallel (~> 1.10)
      parser (>= 2.7.0.1)
      rainbow (>= 2.2.2, < 4.0)
      regexp_parser (>= 1.7)
      rexml
      rubocop-ast (>= 0.0.3)
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 1.4.0, < 2.0)
    rubocop-ast (0.0.3)
      parser (>= 2.7.0.1)
    ruby-macho (1.4.0)
    ruby-progressbar (1.10.1)
    rubyzip (2.3.0)
    sawyer (0.8.2)
      addressable (>= 2.3.5)
      faraday (> 0.8, < 2.0)
    security (0.1.3)
    signet (0.14.0)
      addressable (~> 2.3)
      faraday (>= 0.17.3, < 2.0)
      jwt (>= 1.5, < 3.0)
      multi_json (~> 1.10)
    simctl (1.6.8)
      CFPropertyList
      naturally
    slack-notifier (2.3.2)
    terminal-notifier (2.0.0)
    terminal-table (1.8.0)
      unicode-display_width (~> 1.1, >= 1.1.1)
    thor (0.20.3)
    thread_safe (0.3.6)
    tty-cursor (0.7.1)
    tty-screen (0.8.1)
    tty-spinner (0.9.3)
      tty-cursor (~> 0.7)
    tzinfo (1.2.5)
      thread_safe (~> 0.1)
    uber (0.1.0)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.7)
    unicode-display_width (1.7.0)
    word_wrap (1.0.0)
    xcode-install (2.6.6)
      claide (>= 0.9.1, < 1.1.0)
      fastlane (>= 2.1.0, < 3.0.0)
    xcodeproj (1.18.0)
      CFPropertyList (>= 2.3.3, < 4.0)
      atomos (~> 0.1.3)
      claide (>= 1.0.2, < 2.0)
      colored2 (~> 3.1)
      nanaimo (~> 0.3.0)
    xcpretty (0.3.0)
      rouge (~> 2.0.7)
    xcpretty-travis-formatter (1.0.0)
      xcpretty (~> 0.2, >= 0.0.7)

PLATFORMS
  ruby

DEPENDENCIES
  cocoapods (~> 1.8)
  danger-commit_lint
  danger-gitlab (~> 6.0.0)
  danger-swiftformat (~> 0.7.0)
  danger-swiftlint (~> 0.24.0)
  fastlane (~> 2.0)
  fastlane-plugin-simctl
  rubocop (~> 0.85.0)
  xcode-install (~> 2.6.0)

BUNDLED WITH
   2.1.4
```

**Comment by deivid-rodriguez** [comment]
Thanks for the ticket and I'm sorry the release caused you trouble. I'll come back tomorrow to investigate what happened.

**Comment by deivid-rodriguez** [comment]
Hello! :wave:

Thanks for reporting this issue and sorry for the trouble :pray:.

This has a lot of similarities with #4123.

Proper version selection of bundler should have worked here but unfortunately it needs https://github.com/rubygems/rubygems/pull/2583, which was released with rubygems 3.1, but you're using rubygems 3.0.

As per your CI failure in particular, the specific issue is that you're running `bundle install` in frozen mode but `bundle install` caused changes in the lock file, is that the issue?
